presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -128,11 +131,41 @@ private RelationPlan createInsertPlan(Analysis analysis)
 
         TableMetadata tableMetadata = metadata.getTableMetadata(session, target);
 
+        List<String> visibleTableColumnNames = tableMetadata.getVisibleColumnNames();
+        List<ColumnMetadata> visibleTableColumns = tableMetadata.getVisibleColumns();
+
+        RelationPlan plan = createRelationPlan(analysis);
+
+        Optional<List<String>> insertColumns = analysis.getInsertColumns();
+        if (insertColumns.isPresent() && !insertColumns.get().equals(visibleTableColumnNames)) {
+            PlanNode source = plan.getRoot();
+
+            Map<Symbol, Expression> assignments = visibleTableColumns.stream()
+                    .map(column -> symbolAllocator.newSymbol(column.getName(), column.getType()))
+                    .collect(toImmutableMapReversed(tableColumnAsSymbol -> {
+                        int i = insertColumns.get().indexOf(tableColumnAsSymbol.getName());
```

Comment:
This is incorrect. The names created by SymbolAllocator won't necessarily match the original column names. You should think of them as "variables" that reference some input.

Also, `indexOf` may break if the case of the value in insertColumns doesn't match the case of the name in the metadata. Instead, you should turn names into ColumnHandles in the analyzer and record those in the Analysis.

In general, I think what you want here is something like:

```java
List<ColumnHandle> insertColumns = analysis.getInsertColumns();

TableMetadata tableMetadata = metadata.getTableMetadata(session, target);
Map<String, ColumnHandle> columns = metadata.getColumnHandles(session, target);

ImmutableMap.Builder<Symbol, Expression> assignments = ImmutableMap.builder();
for (ColumnMetadata column :  tableMetadata.getVisibleColumns()) {
    Symbol output = symbolAllocator.newSymbol(column.getName(), column.getType());
    int index = insertColumns.indexOf(columns.get(column.getName()));
    if (index < 0) {
        assignments.put(output, new NullLiteral());
    }
    else {
        assignments.put(output, plan.getSymbol(index).toQualifiedNameReference());
    }
}

```